packages/utils/src/modules-sdk/decorators/inject-manager.ts

```diff
@@ -0,0 +1,31 @@
+import { Context, SharedContext } from "@medusajs/types"
+
+export function InjectManager(managerProperty?: string): MethodDecorator {
+  return function (
+    target: any,
+    propertyKey: string | symbol,
+    descriptor: any
+  ): void {
+    if (!target.MedusaContextIndex_) {
+      throw new Error(
+        `To apply @InjectManager you have to flag a parameter using @MedusaContext`
+      )
+    }
+
+    const originalMethod = descriptor.value
+    const argIndex = target.MedusaContextIndex_[propertyKey]
+
+    descriptor.value = async function (...args: any[]) {
+      const context: SharedContext | Context = args[argIndex] ?? {}
+      const resourceWithManager = await (!managerProperty
+        ? this
+        : this[managerProperty])
+
+      context.manager =
+        context.manager ?? (await resourceWithManager.getFreshManager())
```

Comment:
- isn't it located at `context.transactionManager`?
- `getFreshManager` is not async. no need to `await`

Reply:
This isn't for the transaction, there is a separate decorator for that.